sdk/eventhub/Azure.Messaging.EventHubs/perf/Azure.Messaging.EventHubs.Perf/Azure.Messaging.EventHubs.Perf/Scenarios/Standalone/ConsumeEvents.cs

```diff
@@ -0,0 +1,172 @@
+﻿//Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Producer;
+using CommandLine;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime;
+using System.Threading.Tasks;
+
+namespace Azure.Messafging.EventHub.Perf.Scenarios
+{
+    public class ConsumeEvents
+    {
+        private const string _eventHubName = "test";
+
+        // Settings copied from https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-faq#how-much-does-a-single-capacity-unit-let-me-achieve
+        private const int _messagesPerBatch = 100;
+        private const int _bytesPerMessage = 1024;
+        private static readonly byte[] _payload = new byte[_bytesPerMessage];
+
+        public class Options
+        {
+            [Option('c', "clients", Default = 1)]
+            public int Clients { get; set; }
+
+            [Option('p', "partitions", Default = 1)]
+            public int Partitions { get; set; }
+
+            [Option('s', "send")]
+            public int Send { get; set; }
+
+            [Option('v', "verbose", Default = false)]
+            public bool Verbose { get; set; }
+        }
+
+        public static async Task Execute(string[] args)
+        {
+            if (!GCSettings.IsServerGC)
```

Comment:
You need to special-case 1-core machines:

https://github.com/Azure/azure-sdk-for-net/blob/master/common/Perf/Azure.Test.Perf/PerfProgram.cs#L53-L59